jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/CoreContextHandler.java

```diff
@@ -106,24 +109,64 @@ public void initializeDefaults(Attributes attributes)
 
                         //noinspection unchecked
                         java.util.Collection<Path> deployablePaths = (java.util.Collection<Path>)value;
-                        Optional<Path> optionalDir = deployablePaths.stream()
-                            .sorted(PathCollators.byName(true))
+                        List<Path> nominatedDirs = deployablePaths.stream()
                             .filter(Files::isDirectory)
                             .filter(p -> FileID.isExtension(p.getFileName().toString(), "d"))
-                            .findFirst();
+                            .sorted(PathCollators.byName(true))
+                            .toList();
 
-                        if (optionalDir.isPresent())
+                        if (nominatedDirs.size() == 1)
                         {
                             ResourceFactory resourceFactory = ResourceFactory.of(this);
-                            Resource resourceDir = resourceFactory.newResource(optionalDir.get());
+                            Resource resourceDir = resourceFactory.newResource(nominatedDirs.get(0));
                             setBaseResource(resourceDir);
+                            nominatedDirDefined = true;
+                        }
+                        else if (nominatedDirs.size() > 1)
+                        {
+                            /* this is possible on filesystems that allow things like:
+                             *
+                             * webapps/foo.d
+                             * webapps/FOO.D
+                             * webapps/FOO.d
```

Comment:
A "foo.d" directory should never be a base directory.  It should be invisible to the deployer and is only there so that other configuration for "/foo/" or "/foo.war" should be able to put files into that are used by other configuration.